paasta_tools/vitesscluster_tools.py

```diff
@@ -72,21 +662,98 @@ def validate(
             return []
 
 
-def load_vitess_instance_config(
+def generate_vitess_instance_config(
+    instance_config: Dict,
+) -> Dict:
+    # Generate the vitess instance config from the yelpsoa config
+
+    cpus = instance_config.get("cpus")
+    mem = instance_config.get("mem")
+    deploy_group = instance_config.get("deploy_group")
+    zk_address = instance_config.get("zk_address")
+    paasta_pool = instance_config.get("paasta_pool")
+    paasta_cluster = instance_config.get("paasta_cluster")
+    cells = instance_config.get("cells")
+    keyspaces = instance_config.get("keyspaces")
+    region = instance_config.get("region")
+
+    vitess_instance_config = {
+        "namespace": "paasta-vitessclusters",
+        "cpus": cpus,
+        "mem": mem,
+        "min_instances": 1,
+        "max_instances": 1,
+        "deploy_group": deploy_group,
+        "autoscaling": {"setpoint": 0.7},
```

Comment:
are we intending to autoscale this based on CPU usage? or will y'all add a custom autoscaling signal later?

Reply:
Honestly, we didn't hit the resource limits yet on any of these deployments and can use the default one to start with, but make it configurable in yelpsoa so we don't touch paasta_tools as often. But, yes, if needed in future, we might have to use a custom signal for autoscaling

Reply:
got it - there's some pitfalls with {min,max}_instances + instances, so we may not want to hardcode min/max instances if we're also looking to support `instances: X`